### ISSUE TITLE
Collecting data takes long time

### ISSUE THREAD

**Issue by mountaindude** [ProudElm/solaredgeoptimizers#13]
This might very well be an issue at SolarEdge (pretty likely actually..) but still wanted to report it, in case it's something that can be handled well/less well in this component.

The whole thing might just be due to the rather large (?) number of devices and entities. 
On the other hand, for those having multiple inverters there might be way more optimisers, I guess..

**Background**
Trying to collect data from a 72 panel / 46 optimizer SolarEdge system. 

Running HA 2022.11.4 (=latest as of this writing)  in Docker on a rather beefy Linux server. 
Server performance shouldn't  be an issue thus.

**Symptoms/problem**
After installing the extension I see this in the log. 

![image](https://user-images.githubusercontent.com/1029262/203986851-1fa5a50c-f456-4dcf-b754-dca91e901a42.png)

Waiting for 25-30 minutes, then restarted the component.
Got this instead, but the 46 devices did show up now.

![image](https://user-images.githubusercontent.com/1029262/203987147-aba804d8-e500-4323-b8cb-27155ed74db5.png)

![image](https://user-images.githubusercontent.com/1029262/203987288-9984a2e8-5fae-469d-9531-91626b3573e5.png)

Each device is now sloooowly being. populated with lifetime energy, but that's it.
No other data makes it into these devices/entities. 

![image](https://user-images.githubusercontent.com/1029262/203987716-37d45c3f-e38b-415a-88e8-6dc91451d18e.png)

A bunch of errors in the logs though.

![image](https://user-images.githubusercontent.com/1029262/203987874-1dc33118-7a9a-4df2-9966-64df442b20e8.png)

This one is the only one with a traceback:
![image](https://user-images.githubusercontent.com/1029262/203988007-2cfcb8dd-a9d7-418f-a3e2-5cb04f23a499.png)

Looks like most (all?) entities takes a long time updating:
![image](https://user-images.githubusercontent.com/1029262/203988203-143c1fa1-1e25-4435-bc15-aaca440e86b0.png)

All in all - the slow progress means that only a fraction of the entities have been updated during the 15 min sample interval.
Which might also cause problems (depends on how the code is written I guess), if a new extraction run starts even if the previous one hasn't finished yet.

Anyone else seeing this slow extraction of data?

**Comment by mountaindude** [comment]
Btw, I have restarted the component to see if it was just a first/second-time-after-install issue, but no luck.
Same slowness.

**Comment by ProudElm** [comment]
Ai.. Yeah, the way the integration works is kinda slow, so with such a big system, this would be an issue.  Sorry.

I don't now how to streamline it. The way it works it will request per device (a optimizer) the data from solaredge and than it will fill the sensors (not the lifetime).
This will take 10/15 sec normal per optimizer. But this will depend on network speed etc. 

The problem could be that the return file is so large that processing this each time is causing the problem. I don't know (yet) how i can only request the data once and then fill all the devices/sensors with 1 request.. This could solve many performance issues.

**Comment by mountaindude** [comment]
No worries, I suspected something like that.
I never considered my SolarEdge system here particularly big thou.. But I guess we went a bit crazy on the optimisers :)

I might hack away on this too, to see if there's some other way of getting bulk data.
I am mainly interested in updates every few hours or even 1-2 times per day, maybe that makes the whole thing more feasible. We'll see. 

Really annoying that SolarEdge won't make panel-level data available via the APIs, but that's just how things are for now.
